### DISCUSSION TITLE
Auto translate subtitles

### DISCUSSION THREAD

**Discussion by VahidN** [Tyrrrz/YoutubeExplode#721] [General]
Right now this library provides the existing closed caption urls, such as
https://www.youtube.com/api/timedtext?v=5rRF1xXLHvM&caps=asr&opi=112496729&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1687972731&sparams=ip,ipbits,expire,v,caps,opi,xoaf&signature=23C5DE3AF0303F780043506702490CBFA6B1162D.3EDB0FC0DE34A35649A3962ACF8000AC54C57D0A&key=yt8&kind=asr&lang=en

If we add `&fmt=json3&xorb=2&xobt=3&xovt=3&tlang=fa` to the end of it (or even `&fmt=json3&tlang=fa`) (`fa` is a non-existing `track.Language.Code`):

https://www.youtube.com/api/timedtext?v=5rRF1xXLHvM&caps=asr&opi=112496729&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1687972731&sparams=ip,ipbits,expire,v,caps,opi,xoaf&signature=23C5DE3AF0303F780043506702490CBFA6B1162D.3EDB0FC0DE34A35649A3962ACF8000AC54C57D0A&key=yt8&kind=asr&lang=en&fmt=json3&xorb=2&xobt=3&xovt=3&tlang=fa

It will generate an auto translated version of it with a .json format which is not available in the provided list of existing subtitles.
It would be nice to have a feature for retrieving and processing these types of  subtitles too.

**Comment by Tyrrrz** [comment]
Thanks for the tip. Not sure if I'll include it, but you could just create a `ClosedCaptionTrackInfo` for yourself:

```csharp
var track = new ClosedCaptionTrackInfo(someOtherTrack.Url + "&fmt=json3&tlang=fa", new Language("fa", "Persian"), true);
```

**Comment by VahidN** [comment]
Thanks. Also including `&fmt=json3&` is not required and if we omit it, it will produce an xml file.
